shared/src/price_estimate.rs

```diff
@@ -137,6 +138,29 @@ impl UniswapPriceEstimator {
         })
     }
 
+    /*
+        Returns a vector of (rational) prices for the given tokens
+        denominated in denominator_token.
+    */
+    pub async fn best_execution_spot_prices(
+        &self,
+        tokens: &[H160],
+        denominator_token: H160,
+    ) -> Result<Vec<BigRational>> {
```

Comment:
I'd think floats here is fine since we only use it for one multiplication (it also be slightly more consistent since the functions almost sound the same). I used BigRational to avoid compounding rounding errors when multiplying reserve ratios. Otherwise we could also make the other method return Rationals and do the float conversion in the callsites.

Reply:
> Otherwise we could also make the other method return Rationals and do the float conversion in the callsites.

I'd prefer this way since there is no information loss in case we want to use rationals (as we will in the second part of this PR [here](https://github.com/gnosis/gp-v2-services/pull/373)).

Should I do it in this PR?

Reply:
Nevermind, it actually doesn't matter - there's no info loss. I'll do it here.

Reply:
oh wait, there is because the float is only 64 bits. Hmm, I think I prefer to do everything in rationals and convert to float in the call sites then.

Reply:
Done in 237ed55.